### ISSUE TITLE
[FEATURE REQUEST] New integration 'homematicip_local'

### ISSUE THREAD

**Issue by hvorragend** [xaviml/controllerx#546]
# Feature Request

### Is your feature request related to a problem?

There is a new Homematic integration available: **homematicip_local** 

- https://github.com/danielperna84/custom_homematic
- https://github.com/danielperna84/hahomematic

And the events have changed, so that the old (and still working) homematic-integration in ControllerX doesn't work.

### Describe the solution / feature you'd like

Could you please duplicate the homematic integration and change the event data?

- Instead of 'name' there is the device_id necessary.
- Instead of 'channel' we must now use 'subtype'.
- Instead of 'PRESS_SHORT" (and others) we must capture 'press_short'

~~I don't exactly know if the following lines are mandatory. I will find this out soon.~~

~~- interface_id~~
~~- address~~
~~- name~~

Update: As we can see [here](https://github.com/danielperna84/custom_homematic/blob/devel/blueprints/community/homematicip_local-actions-for-4-button-fm.yaml), we just need the **device_id**, **type** and **subtype**.

It is not possible the examine the events of all devices again. Just the event-data-mapping.

### Additional context

Please see the new events:

```
event_type: homematic.keypress
data:
  interface_id: Homematic-BidCos-RF
  address: LEQ1111111
  type: press_short
  subtype: 2
  device_id: d6ae82a0e7a3caa694cbad346a7aa7bca
  name: Button Interface HM-PBI-4-FM
origin: LOCAL
time_fired: "2022-08-09T18:24:59.843488+00:00"
```

```
event_type: homematic.keypress
data:
  interface_id: Homematic-BidCos-RF
  address: LEQ1111111
  type: press_long
  subtype: 1
  device_id: d6ae82a0e7a3caa694cbad346a7aa7bca
  name: Button Interface HM-PBI-4-FM
origin: LOCAL
time_fired: "2022-08-09T18:30:36.414371+00:00"
```

**Comment by xaviml** [comment]
Hi @hvorragend ,

Thank you for opening the issue. Are you saying that this integration is not working anymore with ControllerX? https://www.home-assistant.io/integrations/homematic/

Or would you like ControllerX to support this new custom Homematic integration?

Regards,
Xavi M.

**Comment by hvorragend** [comment]
The old (and till now the default) integration is working fine with ControllerX.
But there is another integration that is much more developed and has more features. 
And now we have Homematic-devices and not just entities.

I think that we should have two different integrations for "homematic" and "homematicip_local".

**Comment by xaviml** [comment]
Thank you for the further explanation. Would it work if I create a new "custom" integration? I have been meaning to do it for a while, and this use case seems to be the opportunity for it. Adding a new integration it makes sense if lots of people are using it or it is supported by Home Assistant natevely (for this reason I added homematic). If I now add `homematicip_local`, it might bring confusion for future newcomers looking for a homematic integration. My proposal for you would be to have a configuration where you will be able to define the integration by configuration. Something like:

```yaml
example_app:
  module: controllerx
  class: LightController
  integration:
    name: event
    event_type: homematic.keypress  # This is the event to listen to
    controller_key: device_id  # This is the attribute to listen to. It could also be "name"
    action: "{type}_{subtype}"
  light: light.your_light
  mapping:
    press_short_2: toggle  # The key in here should follow the pattern defined above in "action".
```

This way, you can define your own event type and if the integration ever changes, you will have the freedom to change the configuration without requesting a change in ControllerX.

Hopefully, this would work for you.

Regards,
Xavi M.

**Comment by hvorragend** [comment]
Good idea.

**Comment by xaviml** [comment]
Hi @hvorragend ,

I just released [v4.23.0b1](https://github.com/xaviml/controllerx/releases/tag/v4.23.0b1) with a new Event integration. You can read more about it [here](https://controllerx.netlify.app/controllerx/advanced/event-integration/).

You can now try configure the new homematic event like:

```yaml
example_app:
  module: controllerx
  class: LightController
  controller: d6ae82a0e7a3caa694cbad346a7aa7bca
  integration:
    name: event
    event_type: homematic.keypress  # This is the event to listen to
    controller_key: device_id  # This is the attribute to listen to. It could also be "name"
    action_template: "{type}_{subtype}"
  light: light.your_light
  mapping:
    press_short_2: toggle  # The key in here should follow the pattern defined above in "action".
```

Let me know how it works and if you fins any issues if you try it out.

Cheers,
Xavi M.

**Comment by xaviml** [comment]
The event integration is now ready on [ControllerX v4.23.0](https://github.com/xaviml/controllerx/releases/tag/v4.23.0). I will mark this issue as closed.
